crowbar_framework/app/models/service_object.rb

```diff
@@ -1235,6 +1256,9 @@ def apply_role(role, inst, in_queue)
       @logger.debug "run_order #{run_order.inspect}"
     end
 
+    # save databag with the role removal intention
+    databag.save
```

Comment:
But we should only save if there was a change :-) That's why I suggested earlier to use a variable for this.